### DISCUSSION TITLE
Generate two cross-correlated random fields

### DISCUSSION THREAD

**Discussion by Drake-2006** [GeoStat-Framework/GSTools#369] [Q&A]
Many thanks to the author for creating such excellent packages.

Sorry for the beginner-level question, but after reviewing the documentation, I am still unable to find a method for generating cross-correlated random fields. 

Could you please advise on how to correlate two random fields, RFS1 and RFS2, generated by Gstools?

```
import numpy as np
import gstools as gs

# Mean and variance values of the two random fields
mu1 = 20
var1 = 10
mu2 = 1*10**(-5)
var2 = 1*10**(-6)
x = y = z = np.arange(0, 100, 1)

model1 = gs.Exponential(dim=3, var=var1, len_scale=[20, 20, 2])
model2 = gs.Exponential(dim=3, var=var2, len_scale=[20, 20, 2])

srf1 = gs.SRF(model1, mean=mu1)
srf2 = gs.SRF(model1, mean=mu2)
srf1.set_pos([x, y, z], "structured")
srf2.set_pos([x, y, z], "structured")

srf1(seed=1, store='field1')
srf2(seed=2, store='field2')

RFS1 = srf1['field1']
RFS2 = srf2['field2']
```

**Comment by LSchueler** [comment]
Hi Drake,

thanks for the kind words! Unfortunately, currently GSTools does not support cross-correlated random fields. However, you might be able to use[ spatio-temporal fields](https://geostat-framework.readthedocs.io/projects/gstools/en/latest/examples/09_spatio_temporal/index.html) as cross-correlated fields, if you assume the temporal dimension as the cross-correlation between the spatial fields.

Does that help you in any way?
